nodejs/sqlcommenter-nodejs/packages/sqlcommenter-sequelize/provider/opentelemetry.js

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const {context, defaultSetter} = require('@opentelemetry/api');
+const {HttpTraceContext} = require('@opentelemetry/core')
+
+exports.OpenTelemetry = class OpenTelemetry {
+    static addW3CTraceContext(comments) {
+        let propagator = new HttpTraceContext();
```

Comment:
FYI: OpenTelemetry offers a global propagator, which is configurable by customers: https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-api/src/api/propagation.ts

I'm not sure what the correct approach is here. Perhaps the CloudSQL team can comment on what propagation they expect or support on the back-end.

Reply:
I also tried that approach before, but I avoided it due to the side effect of having to adjust the global default propagator. This is what it would look like:
```javascript
propagation.setGlobalPropagator(new HttpTraceContext());
propagation.inject(context.active(), comments, defaultSetter);
```
Without modifying the global propagator, we would be unsure of what method of propagation would be used, which could be something other than the Cloud SQL's expected W3C Trace Context. If the user hasn't set one before, then calling inject will default to a Noop propagator.